server/controllers/user.js

```diff
@@ -21,3 +21,35 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
 
   res.status(200).json({ users: users });
 });
+
+// @route GET /users/:username
+// @desc Get a user by username
+// @access Private
+exports.getUser = asyncHandler(async (req, res, next) => {
+    const username = req.params.username
+    const user = await User.findOne({ username });
+
+    if (!user) res.status(404).send({ error: "User not found" });
+    else res.status(200).send(user);
+});
+
+// @route PATCH /users/:id
+// @desc Update user
+// @access Private
+exports.updateUser = asyncHandler(async (req, res, next) => {
+    const updates = Object.keys(req.body);
+    const allowedUpdates = ['username', 'email', 'password'];
```

Comment:
I don't want to have to maintain this whenever we need to add fields onto the user, this could be big as an application expands. Mongoose provides methods for updating, I would take a look at those.